### PR TITLE
Build: Publish next and latest npm channels to Github

### LINKED ISSUE #27953: Build: move canary channel/tag from npm to github registry
**What would you like to be added**:
Currently we publish a new package to the canary tag/channel in the npm registry on every commit to master. This is really useful when developing plugins and you want to stay in sync with the latest changes in master. Especially close to shipping a new major/minor release.

To make this possible we have a script that is kind of error prone. The reason is that we need to deprecate/unpublish npm packages in the canary tag/channel before publishing a new package to prevent hitting the npmjs registry limit. At multiple time this logic has failed and as a result the whole build in master will fail.

**Why is this needed**:
Earlier this year @dprokop looked into moving the canary tag/channel from npmjs to the github npm registry but that feature of github was not ready to be used since it was in beta. Now it is released as a stable version and what I can see free for public projects. So the suggestion is to move the canary tag/channel to github npm registry to prevent hitting any upper limit and there for simplify the build pipline in master.

https://github.com/features/packages

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
As a follow up to the canary channel github package publishing to make life easier for devs we should also publish next and latest channels to github so there's no need to edit local npmrc config.

**Which issue(s) this PR fixes**:

Fixes #27953

**Special notes for your reviewer**:
We'll continue to publish these channels to the NPM registry.

### CODE REVIEW

File: scripts/release.star [hunk 1]
```diff
@@ -35,10 +35,8 @@ def release_npm_packages_step(edition, ver_mode):
     if edition == 'enterprise':
         return None
 
-    commands = [
-        './node_modules/.bin/lerna bootstrap',
-        'echo "//registry.npmjs.org/:_authToken=$${NPM_TOKEN}" >> ~/.npmrc',
-    ]
+    commands = []
```
Comment (jackw) [inline scripts/release.star:38]: I'm not sure about this. It feels strange that we have a step that releases the packages but only adds the release-packages.sh to the list of commands if `ver_mode = 'release'`. I'm likely missing something related to releases.

File: scripts/release.star [hunk 2]
```diff
@@ -35,10 +35,8 @@ def release_npm_packages_step(edition, ver_mode):
...
+
     if ver_mode == 'release':
         commands.append('./scripts/build/release-packages.sh ${DRONE_TAG}')
```
Comment (dprokop) [inline scripts/release.star:41]:
```suggestion
    if ver_mode == 'release':
        commands = ['./scripts/build/release-packages.sh ${DRONE_TAG}']
```

**Review by dprokop — APPROVED** [review]
Looks great. Let's wait for @aknuds1 to respond to your comment @jack about `commands`

File: scripts/release.star [hunk 3]
```diff
@@ -35,12 +35,8 @@ def release_npm_packages_step(edition, ver_mode):
     if edition == 'enterprise':
         return None
 
-    commands = [
-        './node_modules/.bin/lerna bootstrap',
-        'echo "//registry.npmjs.org/:_authToken=$${NPM_TOKEN}" >> ~/.npmrc',
-    ]
     if ver_mode == 'release':
-        commands.append('./scripts/build/release-packages.sh ${DRONE_TAG}')
+        commands = ['./scripts/build/release-packages.sh ${DRONE_TAG}']
```
Comment (aknuds1) [inline scripts/release.star:39]: You should define `commands`. Not sure how it currently works.
```suggestion
        commands = ['./scripts/build/release-packages.sh ${DRONE_TAG}']
    else:
        commands = []
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
See a problem with the Starlark code.

**Review by aknuds1 — APPROVED** [review]

**Pull request merged**